File: backend/src/main/java/com/quantum/modmail/auth/dto/RegisterRequest.java
package com.quantum.modmail.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Builder;

@Builder
public record RegisterRequest (
    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    String email,

    @NotBlank(message = "Password is required")
    @Size(min = 6, message = "Password must be at least 6 characters")
    String password
) {
}

File: frontend/features/authentication/types/RegisterRequest.ts
import z from "zod";

const RegisterRequest = z.object({
  email: z.email("Invalid email format").min(1, "Email is required"),
  password: z.string().min(6, "Password must be at least 6 characters long"),
  confirmPassword: z.string().min(6, "Confirm password must be at least 6 characters long"),
});

type RegisterRequest = z.infer<typeof RegisterRequest>;

export default RegisterRequest;